tests/unit/test_platforms.py

```diff
@@ -599,3 +602,33 @@ def test_is_platform_definition_subshell(
         assert err_msg in str(exc.value)
     else:
         assert is_platform_definition_subshell(plat_val) is expected
+
+
+def test_get_platform_from_OrderedDictWithDefaults(mock_glbl_cfg):
+    """Get platform works with OrderedDictWithDefaults.
+
+    Most tests use dictionaries to check platforms functionality.
+    This one was added to catch an issue where the behaviour of
+    dict.get != OrderedDictWithDefaults.get.
```

Comment:
Good idea to link to the issue in the test docstring.

Reply:
> This is a small change with no associated Issue.

Do I need to create an issue?

Reply:
```suggestion
    dict.get != OrderedDictWithDefaults.get.
    See - https://github.com/cylc/cylc-flow/issues/4979
```